ci/build_wheel.sh

```diff
@@ -45,6 +45,10 @@ cd "${package_dir}"
 
 python -m pip wheel . -w dist -vvv --no-deps --disable-pip-version-check
 
+# Fix Python wheel tag to be restricted to CPython and version dependent
```

Comment:
For future reference, we could be more explicit about why this change is needed.

```suggestion
# Because we built the library with the run script and manually copied the shared pybind11
# library, the subsequent py_project.toml wheel build was as a pure Python package and results in
# tags that incorrectly indicate it is a universal wheel. To fix this, we need to modify the wheel
# to have CPython ABI and Python tags matching the version of Python that the Python bindings were
# built with.
```